localstack/services/s3/s3_listener.py

```diff
@@ -508,8 +508,13 @@ def remove_xml_preamble(response):
 # --------------
 def get_lifecycle(bucket_name):
     bucket_name = normalize_bucket_name(bucket_name)
+    exists, code, body = is_bucket_available(bucket_name)
+    if not exists:
+        return requests_response(body, status_code=code)
+
     lifecycle = BUCKET_LIFECYCLE.get(bucket_name)
     status_code = 200
+
     if not lifecycle:
         # TODO: check if bucket actually exists
```

Comment:
Can we move lines 511-513 to within the `if`-branch in this line, and remove the `TODO` statement?